editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -2738,42 +2739,74 @@ void VisualShaderEditor::_port_edited(const StringName &p_property, const Varian
 	undo_redo->commit_action();
 }
 
+void VisualShaderEditor::_port_default_popup_input(const Ref<InputEvent> &p_event) {
+	if (p_event.is_null()) {
+		return;
+	}
+
+	if (p_event->is_action_pressed("ui_accept")) {
+		property_editor_popup->hide();
+	}
+}
+
 void VisualShaderEditor::_edit_port_default_input(Object *p_button, int p_node, int p_port) {
 	VisualShader::Type type = get_current_shader_type();
 	Ref<VisualShaderNode> vs_node = visual_shader->get_node(type, p_node);
 	Variant value = vs_node->get_input_port_default_value(p_port);
 
 	edited_property_holder->set_edited_property(value);
+	editing_node = p_node;
+	editing_port = p_port;
 
 	if (property_editor) {
 		property_editor->disconnect("property_changed", callable_mp(this, &VisualShaderEditor::_port_edited));
 		property_editor_popup->remove_child(property_editor);
 	}
 
 	// TODO: Define these properties with actual PropertyInfo and feed it to the property editor widget.
-	property_editor = EditorInspector::instantiate_property_editor(edited_property_holder.ptr(), value.get_type(), "edited_property", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NONE);
-	if (property_editor) {
-		property_editor->set_object_and_property(edited_property_holder.ptr(), "edited_property");
-		property_editor->update_property();
-		property_editor->set_name_split_ratio(0);
-		property_editor_popup->add_child(property_editor);
+	property_editor = EditorInspector::instantiate_property_editor(edited_property_holder.ptr(), value.get_type(), "edited_property", PROPERTY_HINT_NONE, "AAAAAA", PROPERTY_USAGE_NONE, true);
```

Comment:
What is "AAAAAA" for?

Reply:
Getting ready for next-next-next-gen AAA games.

Reply:
Oops :)